ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java

```diff
@@ -38,41 +43,63 @@
  * duration for the command to be executed remotely if configured with a
  * {@code distributedCmdResponseTimeout}.
  */
-public class DistributingExecutor implements StatementExecutor<Statement> {
+public class DistributingExecutor {
 
   private final CommandQueue commandQueue;
   private final Duration distributedCmdResponseTimeout;
   private final BiFunction<KsqlExecutionContext, ServiceContext, Injector> injectorFactory;
   private final KsqlAuthorizationValidator authorizationValidator;
+  private final RequestValidator requestValidator;
 
   public DistributingExecutor(
       final CommandQueue commandQueue,
       final Duration distributedCmdResponseTimeout,
       final BiFunction<KsqlExecutionContext, ServiceContext, Injector> injectorFactory,
-      final KsqlAuthorizationValidator authorizationValidator
+      final KsqlAuthorizationValidator authorizationValidator,
+      final RequestValidator requestValidator
   ) {
     this.commandQueue = Objects.requireNonNull(commandQueue, "commandQueue");
     this.distributedCmdResponseTimeout =
         Objects.requireNonNull(distributedCmdResponseTimeout, "distributedCmdResponseTimeout");
     this.injectorFactory = Objects.requireNonNull(injectorFactory, "injectorFactory");
     this.authorizationValidator =
         Objects.requireNonNull(authorizationValidator, "authorizationValidator");
+    this.requestValidator =
+        Objects.requireNonNull(requestValidator, "requestValidator");
   }
 
-  @Override
   public Optional<KsqlEntity> execute(
       final ConfiguredStatement<Statement> statement,
+      final KsqlParser.ParsedStatement parsedStatement,
       final Map<String, Object> mutableScopedProperties,
+      final String sql,
       final KsqlExecutionContext executionContext,
-      final ServiceContext serviceContext) {
+      final ServiceContext serviceContext,
+      final TransactionalProducer transactionalProducer
+  ) {
     final ConfiguredStatement<?> injected = injectorFactory
         .apply(executionContext, serviceContext)
         .inject(statement);
 
     checkAuthorization(injected, serviceContext, executionContext);
 
     try {
-      final QueuedCommandStatus queuedCommandStatus = commandQueue.enqueueCommand(injected);
+      transactionalProducer.begin();
+      
+      // Don't perform validation on Terminate Cluster statements
+      if (statement.getStatement() instanceof TerminateCluster) {
```

Comment:
shouldn't this be `if (!(statement.getStatement() instanceof TerminateCluster))`

Reply:
Good catch, although this actually doesn't work. The TerminateCluster statement is actually transformed into a TerminateQuery{queryId=CLUSTER} statement after being prepared by the engine. I'm going to need to rethink how to do this.